runtime/vm/monhelpers.c

```diff
@@ -150,6 +107,41 @@ objectMonitorExit(J9VMThread* vmStruct, j9object_t object)
 				}
 #endif /* JAVA_SPEC_VERSION >= 24 */
 			}
+		} else {
+			j9objectmonitor_t newLock = 0;
+			I_32 casSuccess = FALSE;
+			if (0x00 == count) {
+				/* just release the flatlock */
```

Comment:
All comments should be formatted correctly: start with upper case and end with a period.